design/specs/listing-views.md

```diff
@@ -1,27 +1,124 @@
+- Users can clear the search by clearing the input box content or clicking on the `Clear Search` link displayed along with the results summary.
+
+#### Scenario 4b: From the schema explorer view's sidebar
+
+- The user opens the schema explorer by navigating to it via the sidebar or by URL pointing to that object.
+- The user enters a search term in the search input box located at the top of the sidebar list.
+- The list is filtered to show objects whose name matches the search term.
+  - If no results are found, the system displays a notice to indicate the lack of results and potential causes.
+- Users can clear the search by clearing the input box content or clicking on the `Clear Search` link displayed along with the results summary.
+
+## Views
+
+### Schema Details Page
+
+The schema details page will be shown when a schema is open, and no tables are selected. It will contain the schema's name, the total number of tables and views, and additional sections that we can use to present the user with relevant information or links. Examples of content that could go into this section are:
+
+- Recently opened views or tables
+- Activity summary for recent events
+- Getting started options for common actions
+
+## Components
+
+### Schema Explorer Sidebar
+
+The schema explorer sidebar contains the entire list of objects for any schema available in Mathesar. From the sidebar the user can open any object by clicking on its navigation link. The sidebar also includes an input for searching through the listed objects.
+
+### Sidebar Filters
+
+Sidebar filters will allow users to filter the sidebar's contents to show only a particular object type. When set to ' All ', the content will include both tables and views. Setting the filter to `Table` will display only table objects, while `View` will display only view objects.
+
+### Sidebar Sort Control
+
+The sidebar sort control will allow users to select the sorting criteria for the objects displayed in the sidebar.
+
+## Other Considerations
+
+### Navigation Structure
+
+Users of Mathesar will be able to connect multiple databases containing multiple schemas and navigate between them. This prototype assumes there's a single schema and database. For that reason, the interface presented is limited in functionality.
 
-### User sees a default view when they open a schema
-A view can be set as the default that a user will see when they open a schema. Users can set the default view from the 'View Options' menu in a view.
+### Regular vs. Materialized Views
 
-### User can see a list of views referencing a table
-A user from a table can access a list of views referencing that table. This menu also contains a brief description of how users should use views. 
\ No newline at end of file
+No differentiation has been introduced for materialized views, as we concluded that the functionality would be similar. A different icon was proposed, which can be added to the UI improvements design work.
```

Comment:
I apologize for not catching this earlier.  Materialized views are more similar to tables than to non-materialized views.  In fact, they _are_ tables, just annotated with the query that created them as well as enhanced with some functionality to update or refresh them.

From a list of materialized views, I think that it should be possible to:
- see if a materialized view is being (non-concurrently) refreshed from the listing pane, since that would produce a lock on the view making it unavailable.
- trigger a refresh of all materialized views in some list by clicking something.
- long-term, we need to show the last time each materialized view was refreshed in the list, since this is an important piece of info when deciding whether to interact with (refresh) the view.

To reiterate, the distinction between materialized and non-materialized views is crucial, since the non-materialized ones are stateless (I discussed this in another comment) and the materialized ones are stateful.  This leads to quite different use cases, as different as between tables and stateless views.

Reply:
> the non-materialized ones are stateless (I discussed this in another comment) and the materialized ones are stateful

I take it that a materialized view's state is only how out-of-date it is. You can't actually mutate the view without mutating what it references, or am I wrong?

Reply:
@mathemancer The product goals for the alpha release are to ensure that materialized views that are already present in a database are shown as such to the user. We will not have any other functionality for creating or working with materialized views from the UI.

I don't think it is worth spending much design effort into materialized views until we have better support for them in the product.

Reply:
@dmos62 The state of the materialized view is the data.  It's "frozen in time", and maintains its own copy.  The "last updated" part is metadata.  As an example, if you create a materialized view, then change the table it was made from, the materialized view will _not_ change to show the update, until you refresh it.

@kgodey Regardless, we'll be missing functionality that we have planned for both views and tables (e.g., inserting, updating) for materialized views, which can't be altered _except_ by refreshing them.  Even if we don't want to include refresh functionality, I maintain they're as different from normal views as normal views are from tables, and should be separated.  It's unfortunate that they have the name "view", since they're not really the same.

Reply:
@mathemancer we had a conversation about updating and inserting into views in https://github.com/centerofci/mathesar/issues/834 and came to the conclusion that we would have pretty much identical functionality for both views and materialized views.

Having materialized views be a separate section has major product implications – we would then need to explain to non-technical users what this third type of object is and what it's meant to be used for. Since they are going to have pretty much identical functionality to views, I'd rather lump them into views with a different icon and an explanation that they're not up to date when opened (or something similarly basic). The other alternative I can see is to not support them at all or show them in the UI, but that seems worse when we'll already have most of the work done to support them.